pkg/action/executor/k8s_controller.go

```diff
@@ -172,6 +192,47 @@ func (c *parentController) update(updatedSpec *k8sControllerSpec) error {
 	return err
 }
 
+func (c *parentController) rolloutDeploymentConfig(dc *osv1.DeploymentConfig) error {
+	// This will fail if the DC is already paused because of whatever reason
+	name := c.obj.GetName()
+	ns := c.obj.GetNamespace()
+	glog.V(3).Infof("Starting (Instantiating) the deploymentconfig %s/%s rollout", ns, name)
+	deployRequest := osv1.DeploymentRequest{Name: name, Latest: true, Force: true}
+	_, err := c.clients.osClient.AppsV1().DeploymentConfigs(ns).
+		Instantiate(context.TODO(), name, &deployRequest, metav1.CreateOptions{})
+	if err == nil {
+		glog.V(3).Infof("Rolled out (Instantiated) the deploymentconfig %s/%s", ns, name)
+	}
+
+	return err
+}
+
+func (c *parentController) shouldRolloutDeploymentConfig() (bool, *osv1.DeploymentConfig, error) {
+	objName := c.obj.GetNamespace() + "/" + c.obj.GetName()
+	objKind := c.obj.GetKind()
+	if objKind != util.KindDeploymentConfig {
+		return false, nil, nil
+	}
+
+	typedDC := osv1.DeploymentConfig{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(c.obj.Object, &typedDC); err != nil {
+		return false, nil, fmt.Errorf("error converting unstructured %s %s to typed %s: %v", objKind, objName, objKind, err)
+	}
+
+	spec := typedDC.Spec
+	// if no triggers/empty triggers are set ||
+	// if only imageChange trigger and no other trigger is set
+	shouldRollout := (len(spec.Triggers) == 0) ||
+		(len(spec.Triggers) == 1 && spec.Triggers[0].Type == osv1.DeploymentTriggerOnImageChange)
```

Comment:
@irfanurrehman I found there are two kinds of trigger types, I don't quite understand why we only care about `DeploymentTriggerOnImageChange`.   Is it possible these 2 kinds of trigger types exist at the same time but somehow`DeploymentTriggerOnImageChange` is on Triggers[1] rather than Triggers[0]? If yes, do we need a rollout for that case?
```
const (
	// DeploymentTriggerOnImageChange will create new deployments in response to updated tags from
	// a container image repository.
	DeploymentTriggerOnImageChange DeploymentTriggerType = "ImageChange"
	// DeploymentTriggerOnConfigChange will create new deployments in response to changes to
	// the ControllerTemplate of a DeploymentConfig.
	DeploymentTriggerOnConfigChange DeploymentTriggerType = "ConfigChange"
)
```

Reply:
@kevinwangcn (as listed on line 223 and 224) we want to forcefully rollout when triggers are explicitly set to `[]` or there is only 1 trigger which is of type `ImageChange`, because automatic rollout on spec change does not happen in these two cases. As a resize action changes the spec, we want to force a rollout if we can determine that a rollout will not happen. This leaves only one corner case this code does not check, where somebody has specified more then 1 triggers and all of them of type `ImageChange`. I have knowingly not handled that case as our main intention and ask is actually only for those trigger cases where it is specified as `[]`. We do not want to invest too much on this feature and try to handle all corner cases.

Reply:
Yes,  having more than one `ImageChange` shouldn't be a normal case.  It makes sense to me.  Thanks for the explanation @irfanurrehman.